test/conformance/filesystem_perm_test.go

```diff
@@ -0,0 +1,67 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/test"
+)
+
+func verifyPermString(resp string, expected string) error {
+	if len(resp) != len(expected) {
+		return fmt.Errorf("Length of expected and response string don't match. Expected : %d, Response : %d", len(expected), len(resp))
+	}
+
+	for index := range expected {
+		if string(expected[index]) != "*" && expected[index] != resp[index]  {
```

Comment:
1. Dump the mismatching index in the error message as well, it might help identifying the error.
2. Put the expected/response values between quotes, so the reader knows exactly where they start/end in the message.

Reply:
Fixed.